distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso

```diff
@@ -193,8 +194,55 @@ type Date
          other hand, the first day of the week is Monday, and week 1 is the week
          containing the first Thursday of the year. Therefore it is important to
          properly specify the `locale` argument.
-    week_of_year : Locale.Locale -> Integer
-    week_of_year self locale=Locale.default = Time_Utils.week_of_year_localdate self locale.java_locale
+    week_of_year : (Locale.Locale | Nothing) -> Integer
+    week_of_year self locale=Nothing =
+        if locale.is_nothing then
+            Time_Utils.week_of_year_localdate self
+        else
+            Time_Utils.week_of_year_localdate self locale.java_locale
+
+
+    ## Returns if the date is in a leap year.
+    is_leap_year : Boolean
+    is_leap_year self =
+        year = self.year
+        (year%4 == 0) && ((year%100 != 0) || (year%400 == 0))
+
+    ## Returns the number of days in the year represented by this date.
+    length_of_year : Integer
+    length_of_year self =
+        if self.is_leap_year then 366 else 365
```

Comment:
Cannot we just use [`LocalDate.lengthOfYear`](https://docs.oracle.com/en/java/javase/12/docs/api/java.base/java/time/LocalDate.html#lengthOfYear()) instead? Why roll our own?

Reply:
Sticking with this as otherwise inconsistent with `is_leap_year`, now that it uses GregorianCalendar object.

Reply:
Ah, I didn't notice that `LocalDate.isLeapYear` existed. Well maybe we should actually use that? Sorry for causing trouble with the GregorianCalendar.

I guess maybe best to use just `LocalDate.isLeapYear` and in general this set of functions? If we set the cut-off point for date operations to something around 1600 the gregorian calendar issues will not affect us.